modelscope_agent/rag/reader/image.py

```diff
@@ -0,0 +1,266 @@
+import os
+from abc import abstractmethod
+from pathlib import Path
+from typing import Dict, List, Optional, Type, Union
+
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document, ImageDocument
+
+
+class ImageToTextParser:
+
+    def __init__(self, model: str = ''):
+        self.model = model
+
+    @abstractmethod
+    def generate(self, image: Path, prompt: str) -> str:
+        pass
+
+
+class OpenaiAPIParser(ImageToTextParser):
+
+    def __init__(
+        self,
+        model: str = 'qwen-vl-max',
+        base_url: str = 'https://dashscope.aliyuncs.com/compatible-mode/v1',
+        api_key: str = os.getenv('DASHSCOPE_API_KEY', '')):  # noqa
+        from openai import OpenAI
+
+        assert len(api_key), 'api_key is not set.'
```

Comment:
all api_key shall allow user pass in during runtime, no assertion in init stage.